server/container/guice/protocols/jmap/src/main/java/org/apache/james/jmap/JMAPModule.java

```diff
@@ -146,29 +142,28 @@ public RequiredCapabilitiesPrecondition(MailboxManager mailboxManager) {
         }
 
         @Override
-        public void initModule() {
+        public CheckResult check() {
             Preconditions.checkArgument(mailboxManager.hasCapability(MailboxManager.MailboxCapabilities.Move),
-                    "MOVE support in MailboxManager is required by JMAP Module");
+                "MOVE support in MailboxManager is required by JMAP Module");
```

Comment:
Don't you think we should return a bad check in this case? instead of throwing exceptions?

Reply:
I may forgot to implement catching exceptions during checking.